src/Text.Analyzers/Core/IdentifiersShouldBeSpelledCorrectly.cs

```diff
@@ -255,21 +263,39 @@ private static void OnCompilationStart(CompilationStartAnalysisContext compilati
                 SymbolKind.Field,
                 SymbolKind.Parameter);
 
-            IEnumerable<CodeAnalysisDictionary> ReadDictionaries()
+            ImmutableArray<CodeAnalysisDictionary> ReadDictionaries()
             {
-                var fileProvider = AdditionalFileProvider.FromOptions(compilationStartContext.Options);
+                var fileProvider = AdditionalFileProvider.FromOptions(context.Options);
                 return fileProvider.GetMatchingFiles(@"(?:dictionary|custom).*?\.(?:xml|dic)$")
-                    .Select(CreateDictionaryFromAdditionalText)
+                    .Select(GetOrCreateDictionaryFromAdditionalText)
                     .Where(x => x != null)
-                    .ToList();
+                    .ToImmutableArray();
+
+                CodeAnalysisDictionary GetOrCreateDictionaryFromAdditionalText(AdditionalText additionalText)
+                {
+                    if (!_textToDictionary.TryGetValue(additionalText, out var dictionary))
+                    {
+                        dictionary = CreateAndAddDictionary(additionalText);
+                    }
+
+                    return dictionary;
+
+                    // Local function to avoid unnecessary lambda alloc in mainline case.
+                    CodeAnalysisDictionary CreateAndAddDictionary(AdditionalText additionalText)
+                    {
+                        var dictionary = CreateDictionaryFromAdditionalText(additionalText);
+                        dictionary = _textToDictionary.GetValue(additionalText, _ => dictionary);
```

Comment:
💡 It should be possible to use `CompilationStartAnalysisContext.TryGetValue`, and use `additionalText.GetText(...)` as the key. Select the lambda you use for parsing according to the file extension (one lambda handles XML files, the other handles everything else). For the result, return a tuple of `(CodeAnalysisDictionary, Exception?)`. The main processing uses the dictionary, and a new analysis handler either as a registered additinoal file action or in compilation end handles the case where the exception is non-null by reporting diagnostics. This eliminates the need to capture the additional file's path in this callback.

Reply:
done.